src/webview.cpp

```diff
@@ -40,6 +40,19 @@ void WebViewForwardMenu::showEvent(QShowEvent *)
     setGeometry(geo);
 }
 
+void WebView::applyCorrectZoomFactor() {
+    auto url = this->url();
+    auto settingsManager = KiwixApp::instance()->getSettingsManager();
+    qreal zoomFactor;
+    QUrlQuery query(url);
+    if (query.hasQueryItem("pattern")) {
```

Comment:
The intent of this check will be more explicit (and the code more readable) if you encapsulate it into a helper function `WebView::isSearchResultsView()` or introduce an auxiliary variable in this function:

```
const bool isSearchResultsView = QUrlQuery(url).hasQueryItem("pattern");
if ( isSearchResultsView ) {
    ...
```

Besides, I still think that relying only on the presence of a parameter named  `pattern` in the URL query string is a weak criterion that can easily break if we happen to introduce another type of a URL with a similar parameter. Probably it makes sense to also check that the host component of the URL ends with ".search".